### DISCUSSION TITLE
Purpose of `watchParams` in Nuxt page component

### DISCUSSION THREAD

**Discussion by carbotaniuman** [nuxt/nuxt#9639] [Questions (Nuxt 2)]
In our codebase, we have some pages that set `watchParams` to false, alongside an explicit router key function. I tried removing this and it broke the page, but I cannot tell what `watchParams` is supposed to do. (Removing both `watchParams` and the key function works, but it triggers an unneeded rerender)

**Comment by kissu** [comment]
Looks like this: https://nuxtjs.org/docs/2.x/components-glossary/pages-watchquery
Maybe some homemade solution or a specific package ib your `package.json`?

**Comment by TheAlexLichter** [comment]
https://github.com/nuxt/nuxt.js/pull/6244 might provide more info ☺️
